src/tmx/TmxUtils/src/kafka/kafka_producer_worker.cpp

```diff
@@ -33,13 +33,27 @@ namespace tmx::utils
     {
     }
 
+    kafka_producer_worker::kafka_producer_worker(const std::string &brokers)
+        : _broker_str(brokers), _run(true)
+    {
+        init_producer();		
+    }
+
     bool kafka_producer_worker::init()
```

Comment:
What is the purpose of having two methods here init and init producer and one calls the other. Is it just to reduce method complexity? Also why are we calling the init_producer in the constructor? Seems like both init producer and init_topic are called by init which should be called either way after the constructor. Am I missing something?

Reply:
It is to support creating producer without a topic, we only want to init the producer only. But the original init() func has both init_producer and init_topic portion. So I split them into small chunk which will allow the orignal init() works  as it, and allows for reuse of init_producer in support of creating a producer without a topic.

Reply:
To be honest I would prefer you create a super class of kafka producer that is kafka_topicless_producer or the other way around. Kafka Topic Producer inherits from Kafka producer. I think this is weird because it is unclear which init methods need to be called for each or am I missing something.

Reply:
Maybe we can table that as a future improvement since the kafka_client class abstracts this  but could you at least explain how a producer with a topic would be created vs a producer without a topic to me. Is the producer with a topic required to initialize the topic while the other producer only initializes the producer? Is that correct?

Reply:
See the latest code. I removed the init_producer from constructor and allow the plugin to call it and give the plugin a bit more control like the consumers calling init().